src/shex-wrapper.js

```diff
@@ -0,0 +1,131 @@
+// shex-wrapper.js
+
+var _ = require('underscore');
+
+var util = require('util');
+var logger = require('./logger');
+var createLm = require('./create-lm');
+var createState = require('./create-state');
+var factory = require('./pipeline-component-factory');
+var wrapperHelper = require('./wrapper-helper');
+
+var N3 = require("n3");
+var jsonld = require("jsonld").promises;
+var shexiface = require("../shex/shexiface");
+
+/**
+ * This module provides a Shex wrapper.  It is responsible for setting up the
+ * component, and then invoking the RDF pipeline factory to complete building
+ * the component and setting up the vni and state metadata.
+ *
+ * @param nodeDef
+ *
+ * @return a promise to create the noflo rdf component
+ */
+module.exports = function(nodeDef) {
+    return factory({
+        inPorts: _.isArray(nodeDef.inPorts) ?
+            nodeDef.inPorts.reduce(function(map, obj) {
+                if (_.isObject(obj)) {
+                    return obj;
+                }
+                return {[obj]: {dataType: obj == 'input' ? 'all' : 'string'}};
+            }, {}) : nodeDef.inPorts || {
+                input: {
+                    dataType: 'all'
+                }
+            }
+    }, {
+        fRunUpdater: _.partial(fRunUpdater, nodeDef)
+    });
+};
+
+/**
+ * RDF Pipeline SHEX wrapper fRunUpdater API as documented here:
+ *    https://github.com/rdf-pipeline/noflo-rdf-pipeline/wiki/Wrapper-API
+ * This component json-ld on the input port and produces json-ld on the output port
+ *
+ * @param nodeDef
+ * @param vni a virtual node instance
+ */
+function fRunUpdater(nodeDef, vni) {
+    // Execute the updater on the VNI context
+    return new Promise(function(resolve) {
+        var input = vni.inputStates('input').data;
+        if (nodeDef.preprocess)
+            resolve(nodeDef.preprocess(input));
+        else
+            resolve(input);
+    }).then(jsonld_to_n3).then(function shexmap(fromGraph) {
+        var makeTargetNode = _.partial(nodeDef.makeTargetNode, fromGraph);
+        return shexiface(fromGraph, nodeDef.myTypeToShape, nodeDef.myBase,
+		        nodeDef.staticBindings, makeTargetNode, nodeDef.targetFixup
+	    ).then(function(dataAndLog) {
+            return dataAndLog.data;
+        });
+    }).then(n3_to_jsonld).then(function(json) {
+        if (nodeDef.postprocess)
+            return nodeDef.postprocess(json);
+        else
+            return json;
+    }).then(function(json) {
+        var groupLm = wrapperHelper.groupLm(vni.inputStates());
+        vni.outputState({data: json, groupLm: groupLm, lm: createLm()});
+    }).catch(function(e) {
+        wrapperHelper.handleUpdaterException(vni, e);
+    });
+};
+
+function jsonld_to_n3(json) {
```

Comment:
So are we putting everything into a single graph, or generating multiple graphs from ShEx?